src/main/drivers/flash_w25n.c

```diff
@@ -361,9 +360,10 @@ bool w25n_identify(flashDevice_t *fdevice, uint32_t jedecID)
     fdevice->geometry.sectorSize = fdevice->geometry.pagesPerSector * fdevice->geometry.pageSize;
     fdevice->geometry.totalSize = fdevice->geometry.sectorSize * fdevice->geometry.sectors;
 
+    const uint32_t managementStartBlock = W25N_BB_MANAGEMENT_START_BLOCK >= 0 ? W25N_BB_MANAGEMENT_START_BLOCK : fdevice->geometry.sectors + W25N_BB_MANAGEMENT_START_BLOCK;
     flashPartitionSet(FLASH_PARTITION_TYPE_BADBLOCK_MANAGEMENT,
-            W25N_BB_MANAGEMENT_START_BLOCK,
-            W25N_BB_MANAGEMENT_START_BLOCK + W25N_BB_MANAGEMENT_BLOCKS - 1);
+            managementStartBlock,
+            managementStartBlock + W25N_BB_MANAGEMENT_BLOCKS - 1);
```

Comment:
Perhaps

```suggestion
            managementStartBlock + W25N_BB_MANAGEMENT_BLOCKS - W25N_BB_MARKER_BLOCKS);
```

OR

```suggestion
            managementStartBlock + W25N_BB_REPLACEMENT_BLOCKS);
```

Reply:
-1 is to get inclusive upper end of interval (end block is last block of partition)